framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeMirror.java

```diff
@@ -681,6 +682,36 @@ public void clearAnnotations() {
         annotations.clear();
     }
 
+    /**
+     * @return whether this type or any component type is a wildcard type for which Java 7 type
+     *     inference is insufficient. See issue 979, or the documentation on AnnotatedWildcardType.
+     */
+    public boolean containsUninferredTypeArguments() {
+        return uninferredTypeArgumentScanner.visit(this);
```

Comment:
This throws a NPE when `this` is a declared type with out any type arguments.  #3044 has a version of the visitor that works correctly.